### PR TITLE
Recover aliases to Java examples

### LINKED ISSUE #1911: Aliases to Java examples need to be preserved to avoid 404s.
_Originally posted by @chalin in https://github.com/open-telemetry/opentelemetry.io/pull/1898#discussion_r1004879890_

### PR DESCRIPTION
**Pull request by chalin**
- Closes #1911

Preview: https://deploy-preview-1910--opentelemetry.netlify.app/docs/instrumentation/java/

Redirect tests:

- Placeholder page:
  - https://deploy-preview-1910--opentelemetry.netlify.app/docs/java/examples
- Aliases:
  - https://deploy-preview-1910--opentelemetry.netlify.app/docs/java/instrumentation_examples
  - https://deploy-preview-1910--opentelemetry.netlify.app/docs/instrumentation/java/instrumentation_examples

### CODE REVIEW

**Review by svrnm — APPROVED** [review]

File: content/en/docs/instrumentation/java/getting-started.md [hunk 1]
```diff
@@ -139,7 +139,7 @@ For more:
 - Run this example with another [exporter][] for telemetry data.
 - Try [automatic instrumentation](../automatic/) on one of your own apps.
 - For light-weight customized telemetry, try [annotations][].
-- Learn about [manual instrumentation][] and try out more [examples](https://github.com/open-telemetry/opentelemetry-java-docs#java-opentelemetry-examples).
+- Learn about [manual instrumentation][] and try out more [examples](../examples/).
```
Comment (chalin) [inline content/en/docs/instrumentation/java/getting-started.md:142]: Note that the local path to `../examples/` now works.

File: layouts/index.redirects [hunk 2]
```diff
@@ -2,11 +2,19 @@
...
+{{ range $alias := . -}}
+  {{ $alias | printf "%-35s" }} {{ $p.RelPermalink }}
 {{ end -}}
+
+{{ with $p.Params.manualLink -}}
+  {{ $p.RelPermalink }} {{ . }} 301!
 {{ end -}}
 
+{{ end }}{{/* with .Aliases */ -}}
+{{ end }}{{/* range $p */ -}}
```
Comment (chalin) [inline layouts/index.redirects:16]: This code is a bit specialized to our current case where we have a placeholder page with aliases. I'll generalize it in a followup PR.

**Comment by chalin** [comment]
Ready for a final review @svrnm et al. Thanks! You can merge once you're satisfied.

As I mention in one of the comments, I'll submit a followup PR with more generalized redirect-rule code. But I need to think a bit more about how best to do this.

File: layouts/index.redirects (hunk 2 above)
Comment (chalin) [reply layouts/index.redirects:16]: I'll track this via #1914.

**Review by cartermp — APPROVED** [review]
LGTM (whitespace changes are fine to leave in IMO)

**Comment by chalin** [comment]
Thanks @cartermp!

**Pull request merged**